### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by HenaryKhuraijam**
### Recipe App
- Use controllers to handle requests and render empty views.
- Use params from browser requests in a safe way.
- Use preprocessed HTML files with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a web app that requires the user to log in.
- Use Devise gem for authentication.
- Limit access to web app resources based on authorization rules.
- Resolve errors

Participants (2)
@vangelif 
@HenaryKhuraijam

### CODE REVIEW

File: app/views/foods/general.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          <tr class="odd:bg-white even:bg-green-100">
+            <td class="border border-slate-300 py-2"><%= food.name %></td>
+            <td class="border border-slate-300"><%= food.quantity %></td>
+            <td class="border border-slate-300">$<%= food.price * food.quantity %></td>
+          </tr>
+        <% end %>
+      </tbody>
+    </table>
+  </div>
+</div>
```
Comment (rloterh) [inline app/views/foods/general.html.erb:27]: Great job so far 👌, however, shopping list appears empty when sample list and then generated.  Let's ensure that shopping list page:
- [x] shows the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user)
- [x] counts the total food items and total price of the missing food.

**Your shopping list:**        |  **Expected:**
:-------------------------:|:-------------------------:
Home Page | Home Page

File: spec/integration/foods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    it 'renders a button to add food' do
+      expect(page).to have_content('Add Food')
+    end
+
+    it 'redirects to a form for new food' do
+      click_link 'Add Food'
+      expect(page).to have_current_path(new_food_path)
+    end
+  end
+end
```
Comment (rloterh) [inline spec/integration/foods_spec.rb]: - [x] A number of integration specs for food are failing! Can we ensure that all tests are passing?

**Failing integration specs:**
image

File: README.md [hunk 3]
```diff
@@ -1 +1,175 @@
...
+
+I would like to thank [Microverse](https://bit.ly/MicroverseTN)..
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (rloterh) [inline README.md:173]: - [ ] MIT license link is broken. Let's endeavor to fix it to made the ReadMe appear more professional.

**Broken license link:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Henary & Vangelis, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 ⛪🏃 🚜 🏃 🌴 🏃🏫 🌳 🏘️

## Highlights 👍  
- App schema closely follows given ERD!
- Theme looks great so far!
- Added navigation menu for all pages!
- N + 1 problem addressed!
- Unit test framework properly setup!
- Good ReadMe!
- Reviewed projects in turns!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/foods/general.html.erb (hunk 1 above)
Comment (vangelif) [reply app/views/foods/general.html.erb:27]: ## Based on the project requirements, the logged-in user has to add food. The app from the missing food that the user has access to add can then generate the shopping list. 
### These were the requirements for this project. 
### The screenshots below indicate precisely this:

food
recipe
shopping list
requirments

File: spec/integration/foods_spec.rb (hunk 2 above)
Comment (vangelif) [reply spec/integration/foods_spec.rb]: ### All passing now

tests passing

File: app/views/foods/general.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/foods/general.html.erb:27]: @vangelif 

Missing food(s) were not generating shopping list at the time of review. Can kindly you check if the online version of your code is up to date with the requested features? 

Cheers!

**Review by mckent05 — APPROVED** [review]
Hi @HenaryKhuraijam, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job working on your capstone project 👍 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**